homeassistant/components/thermostat/radiotherm.py

```diff
@@ -0,0 +1,127 @@
+    except ImportError:
+        logger.exception(
+            "Error while importing dependency radiotherm. "
+            "Did you maybe not install the radiotherm dependency?")
+        return
+
+    # Detect hosts with hass discovery, config or radiotherm discovery
+    hosts = []
+    if discovery_info:
+        logger.info('hass radiotherm discovery', discovery_info)
+    elif CONF_HOST in config:
+        hosts = [config[CONF_HOST]]
+    else:
+        hosts.append(radiotherm.discover.discover_address())
+
+    name = config.get(CONF_NAME)
+    if hosts is None:
+        logger.error("no radiotherm thermostats detected")
+        return
+
+    tstats = []
+
+    for host in hosts:
+        try:
+            tstat = radiotherm.get_thermostat(host)
+            tstats.append(RadioThermostat(tstat))
+        except (URLError, OSError):
+            logger.exception(
+                "Unable to connect to Radio Thermostat @{}".format(host))
+
+    add_devices(tstats)
+
+
+class RadioThermostat(ThermostatDevice):
+    """ Represent a Radio Thermostat. """
+
+    def __init__(self, device, name=None):
+        self.device = device
+        if name:
+            self.set_name(name)
+        self.set_time()
+
+    @property
+    def name(self):
+        """ Returns the name of the Radio Thermostat. """
+        return self.device.name['raw']
+
+    @property
+    def unit_of_measurement(self):
+        """ Unit of measurement this thermostat expresses itself in. """
+        return TEMP_FAHRENHEIT
+
+    @property
+    def device_state_attributes(self):
+        """ Returns device specific state attributes. """
+        # Move these to Thermostat Device and make them global
+        return {
+            "humidity": None,
+            "target_humidity": None,
+            "fan": self.device.fmode['human'],
+            "mode": self.device.tmode['human']
+        }
+
+    @property
+    def current_temperature(self):
+        """ Returns the current temperature. """
+        return self.device.temp['raw']
+
+    @property
+    def operation(self):
+        """ Returns current operation. head, cool idle """
+        if self.device.tmode['human'] == 'Cool':
+            return STATE_COOL
+        elif self.device.tmode['human'] == 'Heat':
+            return STATE_HEAT
+        else:
+            return STATE_IDLE
+
+    @property
+    def target_temperature(self):
+        """ Returns the temperature we try to reach. """
+
+        if self.operation == STATE_COOL:
+            temp = self.device.t_cool['raw']
+        elif self.operation == STATE_HEAT:
+            temp = self.device.t_heat['raw']
+
+        return round(temp, 1)
+
+    def set_temperature(self, temperature):
+        """ Set new target temperature """
+        if self.operation == STATE_COOL:
+            self.device.t_cool = temperature
+        elif self.operation == STATE_HEAT:
+            self.device.t_heat = temperature
+
+    def set_name(self, name):
```

Comment:
Originally I was specifying the name of the thermostat in configuration.yaml,  but when I started to consider discovery (either with netdisco or radiotherm discovery method) I was unsure how to associate a name with a device that is discovered.   I will just pull it out for now and leave naming of the thermostat outside of hass.